crates/noirc_evaluator/src/ssa/block.rs

```diff
@@ -373,6 +376,34 @@ pub fn rewire_block_left(ctx: &mut SsaContext, block_id: BlockId, left: BlockId)
     }
 }
 
+//replace all instructions in the target block by a false constraint, expect for return instruction
+pub fn short_circuit_inline(ctx: &mut SsaContext, target: BlockId) {
+    // short-circuit the return instruction (if it exists)
+    short_circuit_block(ctx, target);
+    //nop and constrain false
+    let unreachable_op = node::Operation::Constrain(ctx.zero(), None);
+    let unreachable_ins = ctx.add_instruction(Instruction::new(
+        unreachable_op,
+        node::ObjectType::NotAnObject,
+        Some(target),
+    ));
+    let nop = ctx[target].instructions[0];
+    debug_assert_eq!(ctx.get_instruction(nop).operation, node::Operation::Nop);
+    let mut stack = vec![nop, unreachable_ins];
+    //return:
+    let ret_pos = ctx[target].instructions.iter().position(|x| {
+        if let Some(ins) = ctx.try_get_instruction(*x) {
+            return ins.operation.opcode() == Opcode::Return;
+        }
+        false
+    });
+    if let Some(ret) = ret_pos {
+        let ret_id = ctx[target].instructions[ret];
+        stack.push(ret_id);
```

Comment:
Can we move this directly into the check on line 396 instead of checking again?

Reply:
you want to put this code in the closure of position()?

Reply:
Yes, I don't think it is necessary to return the optional position then push to the stack afterward when we can do it inside the closure or as for loop in one step.

Reply:
I go for a for loop then